packages/apollo-engine-reporting/src/plugin.ts

```diff
@@ -114,6 +123,37 @@ export const plugin = <TContext>(
         }
       }
 
+      let shouldReportTrace: null | boolean;
```

Comment:
I'll get into this more below, but unless we're explicitly setting this to `null` for a particular reason, which I don't think we are, let's just use `boolean | undefined` here.

```suggestion
      let shouldReportTrace: boolean | undefined;
```

Reply:
I don't have a strong opinion about null vs undefined but if the choice is null then the variable does need to be initialized.

Alternatively, maybe we can just drop this `let` entirely and use `metrics.captureTraces` directly? Which is even already a `boolean | undefined` (well, it's a `?: boolean` which is close enough).

Reply:
Went with david's suggestion to just use `metrics.captureTraces`.